src/plugins/codeStyle/index.ts

```diff
@@ -167,10 +167,12 @@ export default class CodeStyle {
     validateAAStyle(aa: AALiteralExpression, aaCommaStyle: RuleAAComma, diagnostics: (Omit<BsDiagnostic, 'file'>)[]) {
         const indexes = collectWrappingAAMembersIndexes(aa);
         const last = indexes.length - 1;
+        const singleLine = aa.open.range.start.line === aa.close.range.end.line
+
         indexes.forEach((index, i) => {
             const member = aa.elements[index] as AAMemberExpression;
             const hasComma = !!member.commaToken;
-            if (aaCommaStyle === 'never' || (i === last && aaCommaStyle === 'no-dangling')) {
+            if (aaCommaStyle === 'never' || (i === last && (aaCommaStyle === 'no-dangling') || singleLine )) {
```

Comment:
I think we want to evaluate this condition late instead of always, e.g. `isSingleLine(aa)`
Please run the linter.